core/src/main/scala/io/finch/NegotiateToResponse.scala

```diff
@@ -0,0 +1,48 @@
+package io.finch
+
+import cats.syntax.eq._
+import shapeless._
+
+/**
+  * `NegotiateToResponse` enables content negotiation with client.
+  */
+trait NegotiateToResponse[A, ContentType] {
```

Comment:
On crazy thought. What do people think about building the negotiation into `ToResponse` itself (and avoiding introducing a new abstraction)? I'd imagine, it /would/ be possible to extend `ToResponse` with an extra param `Accept` and make it behaving as `NegotiateToResponse`:


```scala
trait ToResponse[A] {
  type ContentType

  def apply(a: A, cs: Charset, accept: Accept): Response
}
```

So we can derive instances as `ToRespone.Aux[Foo :+: Bar :+: CNil, Application.Json :+: Text.Plain :+: CNil]`, meaning that both `Foo` and `Bar` can be served as JSON and text.

Reply:
I like it in general, but it would mean that anyone who wants to use custom `ToResponse` (i.e. for error rendering) should implement content negotiation on their own.

Reply:
Why would people implement their own `ToResponse`? I'd assume everything can be done at the `Encode` level. `ToResponse` was mostly only needed to accommodate streaming types (that don't necessarily go into `Buf`s).

That said, we might want to add a new `ToResponse` instance if we want to support some new streaming types (i.e., fs2) yet I'm not sure why we'd want to alter the negotiation logic?

Reply:
This is one of the examples I can come up with: https://tools.ietf.org/html/rfc7807
Briefly, to return different `Content-Type` header when there is an error.

`ToResponse` gives more control over the actual response since encoder takes care only of the content/body.

Reply:
I don't think we support that though? `ToService` resolves two `ToResponse` instances of the same content-type (one for success, one for failure).

Reply:
Technically, one can define a function `implicit def tre[CT]: ToResponse[Exception, CT]` and return correct problem json, though this one is resolved for any `CT`